From 6ca8228c993add735d9a962d08b026e5f9ba664d Mon Sep 17 00:00:00 2001
From: Saurav Shah <quic_sauravs@quicinc.com>
Date: Wed, 12 Jun 2024 11:07:51 +0530
Subject: [PATCH] qdl: Remove sahara for pcie channel

Add option for pcie channel.
Over pcie, Sahara will not be used to flash
device programmer.

This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:

Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
SPDX-License-Identifier: BSD-3-Clause-Clear 

Signed-off-by: Saurav Shah <quic_sauravs@quicinc.com>
---
 qdl.c | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/qdl.c b/qdl.c
index e15dcbd..9070cf0 100644
--- a/qdl.c
+++ b/qdl.c
@@ -97,13 +97,13 @@ static void print_usage(void)
 {
 	extern const char *__progname;
 	fprintf(stderr,
-		"%s [--debug] [--storage <emmc|nand|ufs>] [--finalize-provisioning] [--include <PATH>] <prog.mbn> [<program> <patch> ...]\n",
+		"%s [--debug] [--storage <emmc|nand|ufs>] [--finalize-provisioning] [--include <PATH>] [--channel <pcie|usb>] [prog_mbn] [<program> <patch> ...]\n",
 		__progname);
 }
 
 int main(int argc, char **argv)
 {
-	char *prog_mbn, *storage="ufs";
+	char *prog_mbn, *storage="ufs", *channel="usb";
 	char *incdir = NULL;
 	char *serial = NULL;
 	int type;
@@ -118,6 +118,7 @@ int main(int argc, char **argv)
 		{"finalize-provisioning", no_argument, 0, 'l'},
 		{"serial", required_argument, 0, 'S'},
 		{"storage", required_argument, 0, 's'},
+		{"channel", required_argument, 0, 'c'},
 		{0, 0, 0, 0}
 	};
 
@@ -138,6 +139,9 @@ int main(int argc, char **argv)
 		case 'S':
 			serial = optarg;
 			break;
+		case 'c':
+			channel = optarg;
+			break;
 		default:
 			print_usage();
 			return 1;
@@ -150,7 +154,10 @@ int main(int argc, char **argv)
 		return 1;
 	}
 
-	prog_mbn = argv[optind++];
+	if(strcmp(channel, "pcie") != 0)
+	{
+		prog_mbn = argv[optind++];
+	}
 
 	do {
 		type = detect_type(argv[optind]);
@@ -183,10 +190,13 @@ int main(int argc, char **argv)
 	if (ret)
 		return 1;
 
-	qdl.mappings[0] = prog_mbn;
-	ret = sahara_run(&qdl, qdl.mappings, true, NULL, NULL);
-	if (ret < 0)
-		return 1;
+	if(strcmp(channel, "pcie") != 0)
+	{
+		qdl.mappings[0] = prog_mbn;
+		ret = sahara_run(&qdl, qdl.mappings, true, NULL, NULL);
+		if (ret < 0)
+			return 1;
+	}
 
 	ret = firehose_run(&qdl, incdir, storage);
 	if (ret < 0)
-- 
2.17.1

